auth/database/Models.py

```diff
@@ -9,98 +9,143 @@
 import conf as dbconf
 from flaskAlchemyInit import app, db
 
+
 class PermissionEnum(enum.Enum):
     permit = 'permit'
     deny = 'deny'
 
 
-#Model for the database tables
+# Model for the database tables
 class Permission(db.Model):
     __tablename__ = 'permission'
 
     # fields that can be filled by user input
     fillable = ['path', 'method', 'permission']
 
-    #serialise
+    # serialise
     def as_dict(self):
-        return {c.name: str( getattr(self, c.name) ) for c in self.__table__.columns}
+        return {
+                    c.name: str(getattr(self, c.name))
+                    for c in self.__table__.columns
+                }
 
     def safeDict(self):
         return self.as_dict()
 
     id = Column(Integer, primary_key=True, autoincrement=True)
     path = Column(String, nullable=False)
     method = Column(String(30), nullable=False)
-    permission =  Column( Enum(PermissionEnum), nullable=False)
-    users = relationship('User', secondary='user_permission',cascade="delete")
-    groups = relationship('Group', secondary='group_permission',cascade="delete")
+    permission = Column(Enum(PermissionEnum), nullable=False)
+    users = relationship('User', secondary='user_permission', cascade="delete")
+    groups = relationship('Group',
+                          secondary='group_permission',
+                          cascade="delete")
 
 
 class User(db.Model):
     __tablename__ = 'user'
-    # fields that should not be returned to the user
-    sensibleFields = [ 'hash', 'salt', 'secret', 'kongId', 'key' ]
+    # Fields that should not be returned to the user
+    sensibleFields = ['hash', 'salt', 'secret', 'kongId', 'key']
 
-    # fields that can be filled by user input
+    # Fields that can be filled by user input
     fillable = ['name', 'username', 'service', 'email', 'profile']
 
-    #serialise
+    # Serialise class as python dictionary
     def as_dict(self):
         return {c.name: getattr(self, c.name) for c in self.__table__.columns}
 
     def safeDict(self):
-        #don´t serialise sensible fields
-        return {c.name: str( getattr(self, c.name) ) for c in self.__table__.columns \
-            if c.name  not in self.sensibleFields }
+        # Serialise, but drop sensible fields
+        return {
+                c.name: str(getattr(self, c.name))
+                for c in self.__table__.columns
+                if c.name not in self.sensibleFields
+            }
+
+    def getByNameOrID(nameOrId):
+        try:
+            return db.session.query(User).filter_by(id=int(nameOrId)).one()
+        except ValueError:
+            return db.session.query(User).filter_by(username=nameOrId).one()
 
     id = Column(Integer, primary_key=True, autoincrement=True)
     name = Column(String, nullable=False)
     username = Column(String(50), unique=True, nullable=False)
     service = Column(String, nullable=False)
     email = Column(String, nullable=False)
```

Comment:
we could also add an `unique=True` here, right?
I mean, we're checking this ahead of commits, but it's always good to have db schema updated to app constraints, right?